edgedb/protocol/codecs/codecs.pyx

```diff
@@ -798,6 +798,61 @@ cdef pgvector_decode(pgproto.CodecContext settings, FRBuffer *buf):
     return val
 
 
+cdef pgsparse_encode(pgproto.CodecContext settings, WriteBuffer buf,
```

Comment:
It might be worth having a typed memoryview fast path for array/ndarray, like we do for pgvector.

Don't know if there is a way to do that without the annoying code duplication, though